Yosemite/YosemiteTests/Stores/NotificationCountStoreTests.swift

```diff
@@ -0,0 +1,166 @@
+    }
+
+    // MARK: `resetNotificationCount`
+
+    func testResettingNotificationCountOfAGivenTypeReturns0() {
+        let data = NotificationCountPListWrapper(notificationCountBySite: [defaultSiteID: [.comment: 2, .storeOrder: 6]])
+        try! fileStorage.write(data, to: URL(fileURLWithPath: ""))
+
+        var notificationCount: Int?
+        waitForExpectation { expectation in
+            let action = NotificationCountAction.resetNotificationCount(siteID: defaultSiteID, type: .comment) {
+                let loadAction = NotificationCountAction.loadNotificationCount(siteID: self.defaultSiteID, type: .comment) { count in
+                    notificationCount = count
+                    expectation.fulfill()
+                }
+                self.subject.dispatcher.dispatch(loadAction)
+            }
+            subject.dispatcher.dispatch(action)
+        }
+
+        XCTAssertEqual(notificationCount, 0)
+    }
+
+    func testResettingNotificationCountForASiteDoesNotAffectAnotherSite() {
+        // Arrange
+        let anotherSiteID: Int64 = 999
+        let data = NotificationCountPListWrapper(notificationCountBySite: [
+            defaultSiteID: [.comment: 2, .storeOrder: 6],
+            anotherSiteID: [.comment: 3]
+        ])
+        try! fileStorage.write(data, to: URL(fileURLWithPath: ""))
+
+        // Action
+        var notificationCount: Int?
+        waitForExpectation { expectation in
+            let action = NotificationCountAction.resetNotificationCount(siteID: defaultSiteID, type: .comment) {
+                let loadAction = NotificationCountAction.loadNotificationCount(siteID: anotherSiteID, type: .comment) { count in
+                    notificationCount = count
+                    expectation.fulfill()
+                }
+                self.subject.dispatcher.dispatch(loadAction)
+            }
+            subject.dispatcher.dispatch(action)
+        }
+
+        // Assert
+        XCTAssertEqual(notificationCount, 3)
+    }
+
+    // MARK: `resetNotificationCountForAllSites`
+
+    func testResettingNotificationCountForTwoSites() {
+        // Arrange
+        let anotherSiteID: Int64 = 999
+        let data = NotificationCountPListWrapper(notificationCountBySite: [
+            defaultSiteID: [.comment: 2, .storeOrder: 6],
+            anotherSiteID: [.comment: 3]
+        ])
+        try! fileStorage.write(data, to: URL(fileURLWithPath: ""))
+
+        // Action
+        var notificationCountOfDefaultSite: Int?
+        var notificationCountOfAnotherSite: Int?
+        waitForExpectation { expectation in
+            let action = NotificationCountAction.resetNotificationCountForAllSites {
+                let loadDefaultSiteCountAction = NotificationCountAction.loadNotificationCount(siteID: self.defaultSiteID, type: nil) { count in
+                    notificationCountOfDefaultSite = count
+
+                    if notificationCountOfDefaultSite != nil && notificationCountOfAnotherSite != nil {
```

Comment:
Perhaps we can do something like:

1. Add a `count` (default 1) argument in `waitForExpectation` which sets:

    ```swift
        exp.expectedFulfillmentCount = count
    ```
2. Use that in here:

    ```swift
        waitForExpectation(count: 2) { expectation in
    ```
3. Remove this code. 🙂

Reply:
this is a really cool suggestion! updated in https://github.com/woocommerce/woocommerce-ios/pull/2320/commits/829ae5239a4e82485be4811c9bc1c383ba715982